jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/common/Badge.java

```diff
@@ -1,45 +1,103 @@
+import com.epam.jdi.light.angular.elements.enums.BadgePosition;
+import com.epam.jdi.light.angular.elements.enums.BadgeSize;
 import com.epam.jdi.light.common.JDIAction;
 import com.epam.jdi.light.elements.base.UIBaseElement;
 import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.interfaces.base.HasClick;
+import com.epam.jdi.light.elements.interfaces.common.IsText;
+
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.LARGE;
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.MEDIUM;
+import static com.epam.jdi.light.angular.elements.enums.BadgeSize.SMALL;
 
 /**
  * To see an example of Badge web element please visit https://material.angular.io/components/badge/overview.
  */
 
-public class Badge extends UIBaseElement<BadgeAssert> {
+public class Badge extends UIBaseElement<BadgeAssert> implements HasClick, IsText {
 
     @JDIAction("Get '{name}' badge")
-    public UIElement badge() {
-        return this.find(".mat-badge-content");
-    }
-
-    @JDIAction("'{name}' has color '{0}'")
-    public boolean color(String color) {
-        return getColor().equalsIgnoreCase(color);
+    private UIElement badge() {
+        return core().find(".mat-badge-content");
     }
 
     @Override
     public BadgeAssert is() {
         return new BadgeAssert().set(this);
     }
 
-    private String getColor() {
-        return getColor(badge().css("background-color"));
+    @JDIAction("Get '{name}' color")
+    public String color() {
+        return badge().css("background-color");
+    }
+
+    @JDIAction("Get '{name}' size")
+    public BadgeSize size() {
+        if (hasClass("mat-badge-medium")) {
+            return MEDIUM;
+        }
+        if (hasClass("mat-badge-large")) {
+            return LARGE;
+        }
+        if (hasClass("mat-badge-small")) {
+            return SMALL;
+        }
+        return BadgeSize.UNKNOWN;
+    }
+
+    @JDIAction("Get '{name}' vertical position")
+    public BadgePosition positionVertical() {
+        if (hasClass("mat-badge-above")) {
+            return BadgePosition.ABOVE;
+        }
+        if (hasClass("mat-badge-below")) {
+            return BadgePosition.BELOW;
+        }
+        return BadgePosition.UNKNOWN;
     }
 
-    private String getColor(String bgValue) {
-        switch (bgValue) {
-            case "rgba(103, 58, 183, 1)":
-                return "violet";
-            case "rgba(244, 67, 54, 1)":
-                return "red";
-            case "rgba(255, 215, 64, 1)":
-                return "yellow";
-            default:
-                return "unknown color";
+
+    @JDIAction("Get '{name}' horizontal position")
+    public BadgePosition positionHorizontal() {
+        if (hasClass("mat-badge-before")) {
+            return BadgePosition.BEFORE;
+        }
+        if (hasClass("mat-badge-after")) {
+            return BadgePosition.AFTER;
         }
+        return BadgePosition.UNKNOWN;
```

Comment:
By default position is after, so only if we have mat-badge-before it will be BEFORE, in all other variants AFTER

Reply:
still no changes